services/ingestor/src/main/kotlin/company/thewebhook/ingestor/Application.kt

```diff
@@ -1,15 +1,114 @@
 package company.thewebhook.ingestor
 
+import company.thewebhook.eventstore.MessageTooLargeException
+import company.thewebhook.eventstore.Producer
 import company.thewebhook.ingestor.plugins.*
+import io.ktor.http.*
 import io.ktor.server.application.*
+import io.ktor.server.config.*
 import io.ktor.server.netty.*
+import io.ktor.server.plugins.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.ktor.util.*
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.cbor.Cbor
+import kotlinx.serialization.encodeToByteArray
+import org.koin.ktor.ext.inject
 
 fun main(args: Array<String>): Unit = EngineMain.main(args)
 
-fun Application.module() {
+@Serializable
+data class WebhookRequestData(
+    val method: String,
+    val uri: String,
+    val headers: Map<String, List<String>>,
+    val body: String,
+    val receivedAt: Long,
+    val senderHost: String,
+    val serverHost: String,
+)
+
+@ExperimentalSerializationApi
+fun Application.module() = launch {
+    val eventStoreConfig =
+        environment.config
+            .config("event-store.producer")
+            .toMap()
+            .filterValues { it is String }
+            .mapValues { it.value as String }
     configureSerialization()
-    //    configureDatabases()
     configureMonitoring()
     configureHTTP()
+    configureKoin()
     configureRouting()
+    val blacklistedHeaders =
+        environment.config
+            .property("ingestor.blacklisted.headers")
+            .getString()
+            .lowercase()
+            .split(",")
+    val blacklistedHeaderPrefixes =
+        environment.config
+            .property("ingestor.blacklisted.headerPrefixes")
+            .getString()
+            .lowercase()
+            .split(",")
+
+    val producer by inject<Producer<ByteArray>>()
+    producer.connect(eventStoreConfig)
+
+    routing {
+        HttpMethod.DefaultMethods.forEach {
+            route("{...}", it) {
+                handle {
+                    val serverHost = call.request.origin.serverHost
+                    val remoteHost = call.request.origin.remoteHost
```

Comment:
Does this handle reverse proxies and stuff?

Reply:
Doesn't seem so
Ig it's just being used to identify the kafka channel where the webhook is being published
Not sure tho

Reply:
It does indeed handle it. I've installed the XForwardedHeaders plugin: https://ktor.io/docs/forward-headers.html